### PR TITLE
ci: [Automation] Update go matrix

### PR DESCRIPTION
**Pull request by chizhg**
- go matrix updates

Update the go matrix in the CI workflows to include the new supported runtimes.

Auto-generated by https://github.com/chizhg/serverless-runtimes-automation/actions/runs/5901430748

### CODE REVIEW

**Review by jinglundong — APPROVED** [review]

**Comment by kenneth-rosario** [comment]
Lets add cloud.google.com to the harden runner allowlist of the file you modified.
The build pack tests are failing because your are using a forked repository. I will work to exclude them when a pr is from a forked repo.

**Review by sriram-mahavadi — APPROVED** [review]

File: .github/workflows/unit.yml [hunk 1]
```diff
@@ -25,6 +25,13 @@ jobs:
           github.com:443
           objects.githubusercontent.com:443
           proxy.golang.org:443
+          cloud.google.com:443
+          golang.org:443
```
Comment (kenneth-rosario) [inline .github/workflows/unit.yml:29]: Lets use *.golang.org:443 instead of the three different entries on the list.

Comment (sriram-mahavadi) [reply .github/workflows/unit.yml]: no longer needed.

**Review by kenneth-rosario — APPROVED** [review]
Nice, thanks!

**Comment by kenneth-rosario** [comment]
can you rename the PR name to ```ci: [Automation] Update go matrix```

**Pull request merged**